apps/frontend/src/components/Class/Overview/index.tsx

```diff
@@ -1,24 +1,212 @@
+        <p className={styles.userSubmissionLabel}>Description</p>
+        <p className={styles.userSubmissionDescription}>
+          {_class.description ?? _class.course.description}
+        </p>
+        <AttendanceRequirements />
+      </div>
+    </CourseContext.Provider>
+  );
+}
 
+function AttendanceRequirements() {
+  const { data: user } = useReadUser();
+  const { course: _course } = useCourse();
+  const handleFeedbackClick = (e: React.MouseEvent) => {
+    if (!user) {
+      e.preventDefault();
+      const currentPath = window.location.pathname;
+      const redirectPath = `${currentPath}/ratings?feedbackModal=true`;
+      signIn(redirectPath);
+    }
+  };
+  const responses = {
+    Recording: getResponse(_course, MetricName.Recording),
+    Attendance: getResponse(_course, MetricName.Attendance),
+  };
+  const atLeastOneConsensus = Object.values(responses).some(
+    (c) => c == Consensus.Yes || c == Consensus.No
+  );
+  if (!atLeastOneConsensus) {
+    return (
+      <div className={styles.userSubmissionRequirements}>
+        <p className={styles.userSubmissionLabel}>
+          User-Submitted Class Requirements
+        </p>
+        <p className={styles.userSubmissionDescription}>
+          No user-submitted information is available for this course yet.
+        </p>
+        <Link
+          to="ratings?feedbackModal=true"
+          className={styles.suggestEdit}
+          onClick={handleFeedbackClick}
+        >
+          Taken this course? Help others by adding what you know →
+        </Link>
+      </div>
+    );
+  }
   return (
-    <div className={styles.root}>
-      <Details {..._class.primarySection.meetings[0]} />
-      <p className={styles.label}>Description</p>
-      <p className={styles.description}>
-        {_class.description ?? _class.course.description}
+    <div className={styles.userSubmissionRequirements}>
+      <p className={styles.userSubmissionLabel}>
+        User-Submitted Class Requirements
       </p>
-      {_class.course.requirements && (
-        <>
-          <p className={styles.label}>Prerequisites</p>
-          <p className={styles.description}>{_class.course.requirements}</p>
-        </>
-      )}
+      <div>
+        {(() => {
+          switch (responses.Attendance) {
+            case Consensus.Yes:
+              return (
+                <>
+                  <User className={styles.icon} />
+                  <span className={styles.userSubmissionDescription}>
+                    Attendance Required
+                  </span>
+                </>
+              );
+            case Consensus.No:
+              return (
+                <>
+                  <UserXmark className={styles.icon} />
+                  <span className={styles.userSubmissionDescription}>
+                    Attendance Not Required
+                  </span>
+                </>
+              );
+          }
+        })()}
+      </div>
+      <div>
+        {(() => {
```

Comment:
This auto-executing functions are really an anti-pattern here. Same with the switch statements. You have no default for them and you lose efficiency (though that's not really my biggest concern) and readability. I'd recommend the typical `{conditional && <>...</>}` instead